### PR TITLE
add a Wolfram Language notebook to build, install, and test the CodeP…

### PR DESCRIPTION
**Pull request by arnoudbuzing**
…arser paclet.

Tested on Windows, but should work on Linux and macOS.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1 @@
-
-build*
+build
```
Comment (bostick) [inline .gitignore:1]: I don't understand this change. I frequently create directories like build-xcode and build-afl and I don't want any of them to be tracked by git.

File: BuildingCodeParser.nb [hunk 2]
```diff
@@ -0,0 +1,2129 @@
+(* Content-type: application/vnd.wolfram.mathematica *)
```
Comment (bostick) [inline BuildingCodeParser.nb:1]: I don't agree with using StartProcess and ProcessObjects. There are a lot of platform-specific edge cases and there are problems with their implementations.

I would rather documentation for building, installing, and testing be in MarkDown .md files.

File: .gitignore (hunk 1 above)
Comment (arnoudbuzing) [reply .gitignore:1]: Ok, but this also means that any file that starts with "build" or "Build" can not be added here.
Maybe "build*/" would work here? If not, just reject the pull request.

File: BuildingCodeParser.nb (hunk 2 above)
Comment (arnoudbuzing) [reply BuildingCodeParser.nb:1]: That's fine, I can keep this notebook for myself ... I like having a single notebook which builds, installs, and tests everything. Feel free to reject the pull request.

File: .gitignore (hunk 1 above)
Comment (bostick) [reply .gitignore:1]: > Ok, but this also means that any file that starts with "build" or "Build" can not be added here.

Correct; that is the intention

**Comment by kenkangxgwe** [comment]
I thought there is a notebook file at https://github.com/WolframResearch/codeparser/blob/master/Tests/CodeParserNotes.nb. Maybe that one can be refined?

**Comment by bostick** [comment]
I'm closing this PR. We'll go in a different direction.

**Pull request closed without merging**